Allow more granular inclusion of extensions files

kennyadsl opened the pull request:
For example a user could need to include only the rails_helper with its own
spec_helper. This allows doing that with:

```ruby
require 'spec_helper'
require 'solidus_support/extension/rails_helper.rb'
```

It can be more verbose if this is not needed since to fully load the feature_helper it will need:

```ruby
require 'solidus_support/extension/spec_helper.rb'
require 'solidus_support/extension/rails_helper.rb'
require 'solidus_support/extension/feature_helper.rb'
```

Also, this will help if we add more helpers, like `requests_helpers` of others. WDYT?

kennyadsl committed:
Allow more granular inclusion of extensions files

For example a user could need to include only the rails_helper with its own
spec_helper. This allows doing that with:

require 'spec_helper'
require 'solidus_support/extension/rails_helper.rb'

It can be more verbose if this is not needed since to fully load the
feature_helper it will need:

require 'solidus_support/extension/spec_helper.rb'
require 'solidus_support/extension/rails_helper.rb'
require 'solidus_support/extension/feature_helper.rb'

tvdeyen commented:
I like that. 

We could add a new file that includes all of these files at once:

```ruby
require 'solidus_support/testing/setup'
```

Oh, and can we rename that folder? `extension` doesn't sound like something related to tests.

```ruby
require 'solidus_support/testing/spec_helper'
require 'solidus_support/testing/rails_helper'
require 'solidus_support/testing/feature_helper'
```

kennyadsl committed:
Move testing suppoort file from extension to testing folder

This extension is done to support other extensions. There's no need to
specify it. A testing folder sounds more natural.

kennyadsl committed:
Add a testing/all support file that includes all helpers

kennyadsl committed:
Add a Testing Helpers section into README

kennyadsl commented:
@tvdeyen done! Moved all support files into `testing` and I've called the file that includes all of them `testing/all`, does it work?

jhawthorn commented:
The docs are great, but I believe it's a standard pattern for [`rails_helper` to require `spec_helper`](https://github.com/rspec/rspec-rails/blob/master/lib/generators/rspec/install/templates/spec/rails_helper.rb#L2). It doesn't make sense to require one without the other.

The feature_helper _always_ requires the code in the rails_helper, and similar is true for rails_helper and the spec_helper.

kennyadsl commented:
yes but I could want to include a custom spec_helper and not the one provided by this extension, right?

kennyadsl commented:
Making a small recap:

These support files are loaded into extensions to have a standard set of RSpec configuration.
We load those files into the extension's `spec/spec_helper.rb` file. 

If the extension has features spec it is convenient to make:

```ruby
require 'solidus_support/extension/feature_helper'
```

which currently also loads `extension/rails_helper` which loads `extension/spec_helper`.

If the extension does not have features specs we can just make:

```ruby
require 'solidus_support/extension/rails_helper'
```

And it will load test config without features ones.

My concern was more if for example an extension needs the `features_helper` but with something different from what is provided here inside the `rails_helper`. For example it doesn't need to load preferences or factories.

Anyway, if this is not a real usage scenario, I'll be happy to change the PR in order to just add documentation.

tvdeyen commented:
I think what we have right now works good enough. @kennyadsl do you still insist in the change or can we close this for good?

kennyadsl commented:
Yes, we can close this one if we are ok with current version. I'll try to port documentation for current implementation if needed. Thanks both for your feedback.
